src/main/common/maths.c

```diff
@@ -360,3 +361,30 @@ int16_t qMultiply(fix12_t q, int16_t input) {
 fix12_t  qConstruct(int16_t num, int16_t den) {
     return (num << 12) / den;
 }
+
+typedef union
+{
+    float    f;
+    uint32_t l;
+} float_long;
+    
+
+#ifdef USE_THRUST_LINEARIZATION
+// see https://en.wikipedia.org/wiki/Fast_inverse_square_root
+float fast_rsqrt(float number)
```

Comment:
Floating point sqrt is 14 cycles on F4. This code isn't much faster and 1/x is needed to get sqrt (14 cycles)..

Reply:
Wow, thanks! 14 cycles for a straight sqrt is impressive. Happy to change it if you confirm it's as good also on F3. The code originates from SilverWare, probably targeted at stm32f0 which doesn't have a hw float implementation. Maybe that's why the hacky fast_sqrt was used.

Reply:
F3 is Cortex-M4 to, so FPU is identical

Reply:
ok thanks!